### PR TITLE
April 2024 post, Closes #73 

### LINKED ISSUE #73: POST: my "git clean" story

### PR DESCRIPTION
**Pull request by drmowinckels**

### CODE REVIEW

File: content/blog/2024/04-01_git_clean/index.Rmd [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+---
+  
+```{r, include=FALSE}
+options(max.print = 40)
+```
+
+I have a type of love-hate relationship with git.
+While I love the idea of version control, I often find myself in a situation where I have to clean up my git history.
+Maybe I have committed a file that I shouldn't have, or I have made a mistake in my commit message, or I went down a path that just didn't work out.
+In these cases, I find myself using `git clean` or `git chechout` quite a bit.
```
Comment (maelle) [inline content/blog/2024/04-01_git_clean/index.Rmd:23]:
```suggestion
In these cases, I find myself using `git clean` or `git checkout` quite a bit.
```

File: content/blog/2024/04-01_git_clean/index.Rmd [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+In the worst case, I'll even do a `git reset --hard` to go back to a previous commit.
+What all these do, is enable me to go back in time and start over.
+`reset` will let you go back in time even after committing, and deciding that the commit is not what you wanted.
+The other two help more if you haven't committed yet, but have made changes that you want to get rid of.
+
+I have found that the more I use these commands, the more I get used to them.
+But that doesn't mean that I actually do them correctly.
```
Comment (maelle) [inline content/blog/2024/04-01_git_clean/index.Rmd:30]: I love the honesty :joy_cat:

File: content/blog/2024/04-01_git_clean/index.Rmd [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+In these cases, I find myself using `git clean` or `git chechout` quite a bit.
+In the worst case, I'll even do a `git reset --hard` to go back to a previous commit.
+What all these do, is enable me to go back in time and start over.
+`reset` will let you go back in time even after committing, and deciding that the commit is not what you wanted.
+The other two help more if you haven't committed yet, but have made changes that you want to get rid of.
+
+I have found that the more I use these commands, the more I get used to them.
+But that doesn't mean that I actually do them correctly. 
+I have had my fair share of mistakes, where I have lost work that I didn't want to lose.
+I had one when preparing my previous post!
```
Comment (maelle) [inline content/blog/2024/04-01_git_clean/index.Rmd:32]: link to it?

File: content/blog/2024/04-01_git_clean/index.Rmd [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+I was working on the {neuromat} package, and I had made some changes to the code that just wasn't what i was after.
+I decided I wanted to ditch the changes, and go back to what I had before.
+
+`git clean -f`
```
Comment (maelle) [inline content/blog/2024/04-01_git_clean/index.Rmd:37]: explain the f?

File: content/blog/2024/04-01_git_clean/index.Rmd [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+To my horror... I had forgotten to commit the changes I wanted to keep.
+The majority of the good things I had done, went away with the stuff I didn't like. 
+I felt... defeated.
+
+I had to start over, and redo the work I had done.
+[Maëlle Salmon](https://masalmon.eu/) to the rescue!
+I had a chat with her, mostly ranting about how stupid I was, and she suggested I could try to just "undo" in any of my open documents.
```
Comment (maelle) [inline content/blog/2024/04-01_git_clean/index.Rmd:45]: maybe add that "undo" is "Ctrl+Z" in the RStudio IDE?

File: content/blog/2024/04-01_git_clean/index.Rmd [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+In my fury over my git mistakes, I forgot that such a simple solution could exist!
+
+Presto, I could just "undo" myself back to the state I wanted to be in.
+Ok, so I did lose somethings that I had done, but they were rather small and easy to fix.
```
Comment (maelle) [inline content/blog/2024/04-01_git_clean/index.Rmd:49]:
```suggestion
Ok, so I did lose some things that I had done, but they were rather small and easy to fix.
```

File: content/blog/2024/04-01_git_clean/index.Rmd [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+I felt rather stupid.
```
Comment (maelle) [inline content/blog/2024/04-01_git_clean/index.Rmd:51]: Don't write this a second time :wink:

File: content/blog/2024/04-01_git_clean/index.Rmd [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+Both for the mistake, and not thinking of the fairly simple solution.
+
+Thankfully, other more experienced and advanced developers than me also make such mistakes.
+
+{{< toot user="Patricia" server="social.vivaldi.net" toot="112094753393319808" >}}
+
+It's always nice to know that you're not alone in your mistakes.
```
Comment (maelle) [inline content/blog/2024/04-01_git_clean/index.Rmd:58]: link suggestion: https://wizardzines.com/comics/losing-your-work/

**Review by maelle — COMMENTED** [review]
Uninvited comments :innocent:

File: content/blog/2024/04-01_git_clean/index.Rmd (hunk 2 above)
Comment (drmowinckels) [reply content/blog/2024/04-01_git_clean/index.Rmd:30]: well, I strive for full blog honesty :)

File: content/blog/2024/04-01_git_clean/index.Rmd (hunk 7 above)
Comment (drmowinckels) [inline content/blog/2024/04-01_git_clean/index.Rmd:51]:
```suggestion
```

File: content/blog/2024/04-01_git_clean/index.Rmd (hunk 3 above)
Comment (drmowinckels) [inline content/blog/2024/04-01_git_clean/index.Rmd:32]:
```suggestion
I had one when preparing my [previous post](/blog/2024/freesurfer-lmm-r/)!
```

File: content/blog/2024/04-01_git_clean/index.Rmd (hunk 4 above)
Comment (drmowinckels) [inline content/blog/2024/04-01_git_clean/index.Rmd:37]:
```suggestion
`git clean -f`

`git clean -f` removes any untracked files, poof, gone are they. 
And one of the main things I had done was rename and create new files and re-arrange my code.
So, that quickly deleted all of that work. 
Which I kind of wanted, it just went further back in time than I expected.
```

File: content/blog/2024/04-01_git_clean/index.Rmd (hunk 5 above)
Comment (drmowinckels) [inline content/blog/2024/04-01_git_clean/index.Rmd:45]:
```suggestion
I had a chat with her, mostly ranting about how stupid I was, and she suggested I could try to just "undo" in any of my open documents.
In this instance, I was using RStudio IDE, which is my go-to IDE for R work, especially R package dev work (lookout for a future post on my IDE preferences).
"Undo" here meant hitting `ctrl`+ `z`.
```

File: content/blog/2024/04-01_git_clean/index.Rmd (hunk 8 above)
Comment (drmowinckels) [inline content/blog/2024/04-01_git_clean/index.Rmd:58]:
```suggestion
It's always nice to know that you're not alone in your mistakes.

And trust that the awesome [Julia Evans](https://fosstodon.org/@b0rk@jvns.ca) of course has [a zine](https://wizardzines.com/comics/losing-your-work/) for this!
And if she has a zine for it, I know I'm on good company.

And if you, like me (and so many others), need help on your git journey, Maëlle also has an amazing R package to help you practise!
Read her [Instroduction to {saperlipopette}](https://masalmon.eu/2024/01/18/saperlipopette-package-practice-git/) to get started!
```

**Review by drmowinckels — COMMENTED** [review]
Apply Maëlles suggestions

**Review by drmowinckels — COMMENTED** [review]
add suggestions

**Pull request merged**